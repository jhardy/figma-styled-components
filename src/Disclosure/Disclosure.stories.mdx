import { Disclosure } from './Disclosure'
import { Story, Meta } from '@storybook/addon-docs/blocks'

<Meta title='MDX|Disclosure' component={Disclosure} />

# Disclosure

<Story name='Disclosure'>
  <div className='component-box'>
    <Disclosure title='Disclosure'>Disclosure Content</Disclosure>
  </div>
</Story>

<div className="props-table">

## Props

| Prop Name | Type    | Default | Required | Description                           |
| :-------- | :------ | :-----: | :------: | :------------------------------------ |
| title     | string  |    -    |   yes    | Title of the Disclosure               |
| collapsed | boolean |  true   |    -     | If the Disclosure is collapsed or not |

</div>
