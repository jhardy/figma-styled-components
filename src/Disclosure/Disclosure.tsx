import * as React from 'react'
import styled from 'styled-components'
import { Icon } from '../Icon'
import { Text } from '../Text'

export interface DisclosureProps {
  title: string
  collapsed?: boolean
}



class DisclosureFactory extends React.Component<DisclosureProps,{ collapsed: boolean}> {
  constructor (props: DisclosureProps) {
    super(props)
    this.state = {
      collapsed: this.props.collapsed || true
    }

    this.handleClick = this.handleClick.bind(this)
  }

  public render () {
    return(
      <DisclosureWrapper collapsed={this.state.collapsed} {...this.props}>
          <DisclosureHeader onClick={this.handleClick}>
            {this.state.collapsed ? <Icon name='CaretRight' /> : <Icon name='CaretDown' />}
            <Text>{this.props.title}</Text>
          </DisclosureHeader>
          {!this.state.collapsed &&
             <DisclosureContent>
             {this.props.children}
           </DisclosureContent>
          }
      </DisclosureWrapper>
    )
  }

  private handleClick () {
    this.setState({
      collapsed: !this.state.collapsed
    })
  }
}

const DisclosureContent = styled.div`
  padding-left: 22px;
`

const DisclosureHeader = styled.div`
  display: flex;
  align-items: center;
  padding: 8px 0;

  ${Icon} {
    margin-right: 6px;
    margin-top: -3px;
  }
`

const DisclosureWrapper = styled.section<DisclosureProps>``

export const Disclosure = styled(DisclosureFactory)<DisclosureProps>``
