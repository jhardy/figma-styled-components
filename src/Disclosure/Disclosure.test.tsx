import 'jest-styled-components'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Disclosure } from './Disclosure'



test('A default Disclosure', () => {
  const tree = renderer.create(<Disclosure title='Disclosure'>Hello</Disclosure>).toJSON()
  expect(tree).toMatchSnapshot()
})

test('An expaned Disclosure', () => {
  const tree = renderer.create(<Disclosure title='Disclosure' collapsed={false}>Hello</Disclosure>).toJSON()
  expect(tree).toMatchSnapshot()
})
