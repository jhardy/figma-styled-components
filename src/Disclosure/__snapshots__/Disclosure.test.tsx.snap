// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A default Disclosure 1`] = `
.c2 {
  color: currentColor;
  line-height: 0;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 0;
}

.c0 .c1 {
  margin-right: 6px;
  margin-top: -3px;
}

<section
  className=""
  title="Disclosure"
>
  <div
    className="c0"
    onClick={[Function]}
  >
    <div
      className="c1 c2"
    >
      <svg
        fill="none"
        height={16}
        width={16}
      >
        <path
          d="M12 8L8 5v6l4-3z"
          fill="currentColor"
          fillOpacity={0.8}
        />
      </svg>
    </div>
    <span
      className="c3"
    >
      Disclosure
    </span>
  </div>
</section>
`;

exports[`An expaned Disclosure 1`] = `
.c2 {
  color: currentColor;
  line-height: 0;
}

.c3 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  font-family: "Inter",sans-serif;
  color: rgba(0,0,0,0.8);
  font-weight: 400;
  font-size: 11px;
  -webkit-letter-spacing: 0.005em;
  -moz-letter-spacing: 0.005em;
  -ms-letter-spacing: 0.005em;
  letter-spacing: 0.005em;
  line-height: 16px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px 0;
}

.c0 .c1 {
  margin-right: 6px;
  margin-top: -3px;
}

<section
  className=""
  title="Disclosure"
>
  <div
    className="c0"
    onClick={[Function]}
  >
    <div
      className="c1 c2"
    >
      <svg
        fill="none"
        height={16}
        width={16}
      >
        <path
          d="M12 8L8 5v6l4-3z"
          fill="currentColor"
          fillOpacity={0.8}
        />
      </svg>
    </div>
    <span
      className="c3"
    >
      Disclosure
    </span>
  </div>
</section>
`;
